PCSamples/Graphics/SimpleHDR_PC/DeviceResources.cpp

```diff
@@ -550,11 +567,52 @@ void DeviceResources::UpdateColorSpace()
 #if defined(NTDDI_WIN10_RS2)
     if (m_swapChain)
     {
-        ComPtr<IDXGIOutput> output;
-        if (SUCCEEDED(m_swapChain->GetContainingOutput(output.GetAddressOf())))
+        if (!m_dxgiFactory->IsCurrent())
+        {
+            // Output information is cached on the DXGI Factory. If it is stale we need to create a new factory.
+            CreateFactory();
+        }
+
+        // Get the retangle bounds of the app window
+        RECT windowBounds;
+        GetWindowRect(m_window, &windowBounds);
```

Comment:
Technically ``GetWindowRect`` returns a BOOL for success or failure. 

How about:

```
if (!GetWindowRect(m_window, &windowBounds))
    throw DX::com_exception(HRESULT_FROM_WIN32(GetLastError()));
````

> Note in the modern template, I'd actually use this mouthful:  ``throw .std::system_error(std::error_code(static_cast<int>(GetLastError()), std::system_category()), "GetWindowRect");``. These older templates didn't use that pattern or include ``<system_error>``.

> For true cases of "I don't ever care about the return" you can use ``(void)GetWindowRect(...);`` but that upsets some compiler tidy warnings. In the modern template, I'd have used ``std::ignore = GetWindowRect(...``, but this older template doesn't include ``<tuple>``.